src/libs/Navigation/AppNavigator/BaseDrawerNavigator.js

```diff
@@ -66,6 +67,19 @@ class BaseDrawerNavigator extends Component {
                     ),
                     swipeEdgeWidth: 500,
                 }}
+                screenListeners={{
+                    state: (e) => {
+                        const state = e.data.state;
+
+                        // Get the history that has drawer type to get the status.
+                        const hasDrawerHistory = _.find(state.history || [], h => h.type === 'drawer');
+
+                        // hasDrawerHistory will has undefined value if the route drawer is equal to initial route drawer.
```

Comment:
I don't really understand this. What does "if the route drawer is equal to initial route drawer" mean exactly?

Reply:
Also, `hasDrawerHistory` is not a boolean so this variable name should communicate that this is a "first history item that is type drawer"? (If I am understanding this correctly - not really getting any of this code)

Reply:
When we first open the page and shows up the LHN, then click on any report item, when on the report screen the drawer has the history. But when we go back again to LHN, the drawer history will return undefined.

> Also, hasDrawerHistory is not a boolean so this variable name should communicate that this is a "first history item that is type drawer"? (If I am understanding this correctly - not getting any of this code)

Yeah, you're right.

Reply:
Sorry for not making it clear, what I mean by drawer history is history that has the type `drawer`.

<img width="606" alt="Screen Shot 2022-05-20 at 13 12 23" src="https://user-images.githubusercontent.com/25520267/169464379-0b229f5a-87a9-4b73-ad54-c46280008b61.png">

For example on this screenshot, when I first open the app it's showing a report screen. Then I go to LHN, in this step the history will have a history that has the type `drawer` which has `open` value. After that, I open the report screen again and there's no history for type `drawer`.

Reply:
Ok so the history always has a value - it's just not always a type "drawer" history item. That would be good to clarify in the comment.

But maybe let's determine if this is the correct solution first...